spec/tapioca/compilers/dsl/active_record_associations_spec.rb

```diff
@@ -407,7 +407,42 @@ def commenters_attributes=(attributes); end
     end
 
     describe("with errors") do
-      it("generates RBI file for broken has_many :through collection association with errors") do
+      it("generates RBI file for broken has_one association") do
+        add_ruby_file("schema.rb", <<~RUBY)
+          ActiveRecord::Migration.suppress_messages do
+            ActiveRecord::Schema.define do
+              create_table :posts do |t|
+                t.references(:author)
+              end
+            end
+          end
+        RUBY
+
+        add_ruby_file("post.rb", <<~RUBY)
+          class Post < ActiveRecord::Base
+            has_one :author
+          end
+        RUBY
+
+        expected = <<~RBI
+          # typed: strong
+
+          class Post
+            include GeneratedAssociationMethods
+
+            module GeneratedAssociationMethods; end
+          end
```

Comment:
I'm not too sure off hand, I haven't dug into the logic of that. So can't really speak to how difficult it would be.

Reply:
We normally try to avoid this, but in this case, it is fine, since there was an error in `Post` that needs to be fixed. Once the error is fixed, we would either not generate this file at all or will be generating proper association methods for it.